### PR TITLE
 Replace wireframe cube with solid, rotating coloured cube on a green background

### PR DESCRIPTION
**Pull request by josh-disguise**
cubeee

Easier to verify for debugging purposes - moves, no black background, can be picked out in a downscaled image.

### CODE REVIEW

File: src/samples/DX11/DX11.cpp [hunk 1]
```diff
@@ -154,6 +161,29 @@ struct ConstantBufferStruct
     DirectX::XMMATRIX worldViewProjection;
 };
 
+std::mutex logMutex;
+decltype(rs_logToD3)* g_rs_logToD3 = nullptr;
+void rsLog(const char* msg)
+{
+    std::lock_guard<std::mutex> lock(logMutex);
```
Comment (husberg-disguise) [inline src/samples/DX11/DX11.cpp:168]: Shouldn't need a mutex, should it?

File: src/samples/DX11/DX11.cpp [hunk 2]
```diff
@@ -154,6 +161,29 @@ struct ConstantBufferStruct
...
+    g_rs_logToD3(msg);
```
Comment (husberg-disguise) [inline src/samples/DX11/DX11.cpp:169]: Null-check

File: src/samples/DX11/DX11.cpp [hunk 3]
```diff
@@ -285,6 +324,7 @@ int main()
     };
     std::unordered_map<StreamHandle, RenderTarget> renderTargets;
     FrameData frameData;
+    int tick = 0;
```
Comment (husberg-disguise) [inline src/samples/DX11/DX11.cpp:327]: No instance-local variables. All logic must be deterministic and based off the frame request data

File: src/samples/DX11/DX11.cpp [hunk 4]
```diff
@@ -154,6 +161,29 @@ struct ConstantBufferStruct
     DirectX::XMMATRIX worldViewProjection;
 };
 
+std::mutex logMutex;
+decltype(rs_logToD3)* g_rs_logToD3 = nullptr;
+void rsLog(const char* msg)
```
Comment (husberg-disguise) [inline src/samples/DX11/DX11.cpp:166]: Maybe move to another PR

File: src/samples/DX11/DX11.cpp [hunk 5]
```diff
@@ -356,7 +396,7 @@ int main()
                 const RenderTarget& target = renderTargets.at(description.handle);
                 context->OMSetRenderTargets(1, target.view.GetAddressOf(), nullptr);
 
-                const float clearColour[4] = { 0.f, 0.f, 0.f, 0.f };
+                const float clearColour[4] = { 0.f, 0.2f, 0.f, 0.f };
```
Comment (husberg-disguise) [inline src/samples/DX11/DX11.cpp:399]: Not sure about the green tint. May confuse people with alpha/premult alpha blending errors. 

This should be a clear sample of how to do frontplate rendering.

**Review by husberg-disguise — CHANGES_REQUESTED** [review]

**Comment by josh-disguise** [comment]
Will fix and resubmit.

**Pull request closed without merging**